noteable/datasource_postprocessing.py

```diff
@@ -338,10 +339,12 @@ def postprocess_clickhouse(
 ) -> None:
     connect_args = create_engine_kwargs["connect_args"]
 
-    # Pop the keys we want to use to build the query parameters in the connection string
-    # so that they don't get doubly passed down into the `noteable.sql.connection.Connection.set` call
-    # in `noteable.datasources.bootstrap_datasource()`.
-    dsn_dict["query"] = {
-        "protocol": connect_args.pop("protocol"),
-        "verify": str(connect_args.pop("verify")).lower(), # converts True/False to true/false
-    }
+    # These are the enum options from the JSON schema for the dropdown titled "Secure Connection (HTTPS)"
+    # Convert them to the values that the clickhouse driver expects.
+    verify = connect_args["verify"]
+    if verify == "Yes, use HTTPS":
+        connect_args.update({"protocol": "https", "verify": False})
+    elif verify == "Yes, use HTTPS and verify server certificate":
+        connect_args.update({"protocol": "https", "verify": True})
+    elif verify == "No, use HTTP":
+        connect_args.update({"protocol": "http", "verify": False})
```

Comment:
add trailing else clause with logging a huge 'unexpected value' ish error instead of just doing nothing if we get a surprise value.

Plus additional test thereof.